uqcsbot/course_ecp.py

```diff
@@ -0,0 +1,115 @@
+    )
+    @yelling_exemptor(
+        input_args=["course1, course2, course3, course4"]
+    )    
+    
+    async def courseecp(
+        self,
+        interaction: discord.Interaction,
+        course1: str,
+        course2: Optional[str],
+        course3: Optional[str],
+        course4: Optional[str],
+        year: Optional[int] = None,
+        semester: Optional[Offering.SemesterType] = None,
+        campus: Offering.CampusType = "St Lucia",
+        mode: Offering.ModeType = "Internal",
+    ):
+        """
+        Returns the URL of the ECPs for course codes given. Assumes the same semester and year for the course codes given.
+
+        """
+        await interaction.response.defer(thinking=True)
+
+        possible_courses = [course1, course2, course3, course4]
+        course_names = [c.upper() for c in possible_courses if c != None]
+        course_name_urls: dict[str,str] = {}
+        offering = Offering(semester=semester, campus=campus, mode=mode)
+        
+        try:
+            for course in course_names:
+                course_name_urls.update({course: get_course_profile_url(course, offering, year)})
+        except HttpException as exception:
+            logging.warning(
+                f"Received a HTTP response code {exception.status_code}. Error information: {exception.message}"
+            )            
+            await interaction.edit_original_response(
+                content=f"Could not contact UQ, please try again."
+            )
+            return
+        except (CourseNotFoundException, ProfileNotFoundException) as exception:     
+            await interaction.edit_original_response(content=exception.message)
+            return
+        
+        # If year is none assign it the current year
+        if not year:
+            year = datetime.today().year
+        
+        # If semester is none assign it the current estimated semester
+        if not semester:
+            current_month = datetime.today().month
+            if 2 <= current_month <= 6:
+                semester = "1"
+            elif 7 <= current_month <= 11:
+                semester = "2"
+            else:
+                semester = "Summer"
+
+        # Create the embedded message with the course names and details
+        embed = discord.Embed(
+            title=f"Course ECP: {', '.join(course_names)}",
+            description=f"For Semester {semester} {year}, {mode}, {campus}"
+        )
+
+        # Add the ECP urls to the embedded message
+        if course_name_urls:
+            for course in course_name_urls:
+                embed.add_field(
+                    name=f"",
+                    value=f"[{course}]({course_name_urls.get(course)}) ",
+                    inline=False,
+                )
+        else:
+            await interaction.edit_original_response(
+                content=f"No ECP could be found for {course_names}. The {course_names}'s ECP might not be available."
```

Comment:
Nitpick: Potentially this might look weird if multiple courses were searched for. But this error should hopefully not show at all, so feel free to ignore this.